q2_fmt/_peds.py

```diff
@@ -468,3 +484,89 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
 def _mask_recipient(donor_mask, recip_df):
     maskedrecip = donor_mask & recip_df
     return maskedrecip
+
+
+def peds_bootstrap(table: pd.DataFrame, metadata: qiime2.Metadata,
+                   time_column: str, reference_column: str,
+                   subject_column: str,
+                   filter_missing_references: bool = False,
+                   drop_incomplete_subjects: bool = False,
+                   drop_incomplete_timepoint: list = None,
+                   bootstrap_replicates: int = 999) -> (pd.DataFrame):
+
+    metadata_df = metadata.to_dataframe()
+    donor = metadata_df[metadata_df.index.isin(
+        _check_reference_column(metadata=metadata_df,
+                                reference_column=reference_column))]
+
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+    fake_donor = pd.DataFrame([])
+    for i in range(0, bootstrap_replicates+1):
+        if i == 0:
```

Comment:
Bump this out of the `for` loop, so you don't have to do the check every time. Then you can also loop over `range(bootstrap_replicates)`.